packages/build-info/src/detect-build-system.ts

```diff
@@ -0,0 +1,173 @@
+import { readFileSync } from 'fs'
+import path from 'path'
+
+import { findUpSync } from 'find-up'
+import { PackageJson } from 'read-pkg'
+
+export type BuildSystem = {
+  name: string
+  version?: string | undefined
+}
+
+type BuildSystemHandler = (baseDir: string, rootDir?: string) => Promise<BuildSystem | undefined>
+
+export const detectBuildSystems = async (baseDir: string, rootDir?: string): Promise<BuildSystem[]> => {
+  const buildTools = Object.keys(BUILD_SYSTEMS)
+  const buildSystems = await Promise.all(buildTools.map(async (tool) => await BUILD_SYSTEMS[tool](baseDir, rootDir)))
+
+  return buildSystems.reduce((res, tool) => {
+    if (tool) {
+      res.push(tool)
+    }
+
+    return res
+  }, [] as BuildSystem[])
+}
```

Comment:
```suggestion
  return buildSystems.filter(Boolean).reduce((res, tool) => [...res, tool], [] as BuildSystem[])
}
```

Reply:
Filtering doesn't narrow the type from `BuildSystem | undefined` to `BuildSystem`